pythonwhat/test_function.py

```diff
@@ -78,6 +78,14 @@ def test_function(name,
     state.extract_function_calls()
     solution_calls = state.solution_function_calls
     student_calls = state.student_function_calls
+    student_imports = state.student_imports
+
+    # for messaging purposes: replace with original alias or import again.
+    stud_name = name
+    for (short, full) in student_imports.items():
+        if full in stud_name:
+            stud_name = stud_name.replace(full, short)
```

Comment:
I think the idea is good but the implementation is not. 
```
DC_CODE='''
import math as m
import mathplotlib.pyplot as plt
plt.scatter([1,2,3], [1*m.pi, 2*m.pi, 3*m.pi])
'''
DC_SCT='''
test_function("mathplotlib.pyplot.scatter")
'''
```
You're looping over the imports of the student: `["m": "math", "plt": "mathplotlib.pyplot"]`. `student_name == "mathplotlib.pyplot.scatter"` to begin with. Then you check for the first import if `"math" in "mathplotlib.pyplot.scatter"`, which it is. Then you're replacing `"math"` in `"mathplotlib.pyplot.scatter"` by `"m"`. So `stud_name == "mplotlib.pyplot.scatter"`. I didn't test so I could be wrong. 

I would select everything before the last `.` in `stud_name` and use `student_imports` as a hash with what you find.